src/Locale.php

```diff
@@ -183,18 +198,21 @@ private function saveLocale(string $locale, ResponseInterface $response): Respon
         return $cookie->addToResponse($response);
     }
 
+    /**
+     * @return array{0:string, 1:string|null}
+     */
     private function parseLocale(string $locale): array
     {
-        if (str_contains($locale, '-')) {
-            return explode('-', $locale, 2);
-        }
+        $localeParts = preg_split('/[-_]/', $locale, 2, PREG_SPLIT_NO_EMPTY);
```

Comment:
Do we now support both BCP47 and Unix formats?

preg_split is definitely slower than str_contains + explode and early return.